monitors/BalanceMonitor.js

```diff
@@ -0,0 +1,138 @@
+const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
+
+class BalanceMonitor {
+  constructor(logger, tokenBalanceClient, account, botsToMonitor) {
+    this.logger = logger;
+    this.account = account;
+
+    // An array of bot objects to monitor. Each bot's `botName` `address`,
+    // `CollateralThreshold` and`syntheticThreshold` must be given. Example:
+    // [{ name: "Liquidator Bot",
+    //   address: "0x12345"
+    //   collateralThreshold: x1,
+    //   syntheticThreshold: x2,
+    //   etherThreshold: x3 },
+    // ...]
+    this.botsToMonitor = botsToMonitor;
+
+    // Structure to monitor if a wallet address have been alerted yet for each alert type.
+    this.walletsAlerted = {};
+
+    // Populate walletsAlerted for each bot, starting with each alert type at not sent.
+    for (let bot of botsToMonitor) {
+      this.walletsAlerted[bot.address] = {
+        collateralThreshold: false,
+        syntheticThreshold: false,
+        etherThreshold: false
+      };
+    }
+
+    // Instance of the tokenBalanceClient to read account balances from last change update.
+    this.client = tokenBalanceClient;
+    this.web3 = this.client.web3;
+
+    this.formatDecimalString = createFormatFunction(this.web3, 2);
+
+    // TODO: replace this with a fetcher that pulls the actual collateral token symbol
+    this.collateralCurrencySymbol = "DAI";
+    this.syntheticCurrencySymbol = "UMATEST";
```

Comment:
we should be able to get them from the client that provides data fortunately .

Reply:
But only if `update()` has been called on the client before construction, no?

Reply:
ye, you right. Also, the client does not currently store this information; this is why I didnt implement it. What I can do is pull this from the client within the function where it is needed and not initiate this variable in the constructor. This will work because the `update()` function is always called before the modules alert generating function.

Reply:
Seems reasonable. Nothing has to be changed now, just wanted to mention it.